profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PprofAllocationEventExporter.java

```diff
@@ -0,0 +1,291 @@
+            locationTable.get(
+                "", // file name is not known
+                method.getType().getName() + "." + method.getName(),
+                frame.getLineNumber()));
+      }
+    }
+    if (spanContext != null && spanContext.isValid()) {
+      Label.Builder traceIdLabel = Label.newBuilder();
+      traceIdLabel.setKey(stringTable.get("trace_id"));
+      traceIdLabel.setStr(stringTable.get(spanContext.getTraceId()));
+      sample.addLabel(traceIdLabel.build());
+
+      Label.Builder spanIdLabel = Label.newBuilder();
+      spanIdLabel.setKey(stringTable.get("span_id"));
+      spanIdLabel.setStr(stringTable.get(spanContext.getSpanId()));
+      sample.addLabel(spanIdLabel.build());
+
+      Label.Builder eventTypeLabel = Label.newBuilder();
+      eventTypeLabel.setKey(stringTable.get("event_type"));
+      eventTypeLabel.setStr(stringTable.get(event.getEventType().getName()));
+      sample.addLabel(eventTypeLabel.build());
+    }
+    // XXX are there any more attributes that should be added to the sample?
+    // XXX PlainTextAllocationEventExporter adds attributes from the used AllocationEventSampler
+
+    profile.addSample(sample);
+  }
+
+  private byte[] serializePprof() {
+    Profile profile = profileData.profile.build();
+    profileData = new ProfileData();
+
+    try {
+      ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+      boolean gzip =
+          allocationDataFormat == AllocationDataFormat.PPROF_GZIP
+              || allocationDataFormat == AllocationDataFormat.PPROF_GZIP_BASE64;
+      try (OutputStream outputStream = gzip ? new GZIPOutputStream(byteStream) : byteStream) {
+        profile.writeTo(outputStream);
+      }
+      byte[] bytes = byteStream.toByteArray();
+      if (allocationDataFormat == AllocationDataFormat.PPROF_BASE64
+          || allocationDataFormat == AllocationDataFormat.PPROF_GZIP_BASE64) {
+        bytes = Base64.getEncoder().encode(bytes);
+      }
+      return bytes;
+    } catch (IOException exception) {
+      throw new IllegalStateException("Failed to serialize pprof", exception);
+    }
+  }
+
+  public static final AttributeKey<String> DATA_TYPE = stringKey("com.splunk.datatype");
+  public static final AttributeKey<String> ENCODING = stringKey("com.splunk.encoding");
+
+  @Override
+  public void flush() {
+    // Flush is called after each JFR chunk, hopefully this will keep batch sizes small enough.
+    byte[] bytes = serializePprof();
+    String encoding = allocationDataFormat.toString().toLowerCase(Locale.ROOT).replace('_', '-');
+
+    // XXX just to give an overview of exported data size
+    logger.info("Exporting {}, size {}", encoding, bytes.length);
+
+    Attributes attributes =
+        Attributes.builder()
+            .put(SOURCE_TYPE, PROFILING_SOURCE)
+            .put(DATA_TYPE, "allocation") // tells that this message is about allocation samples
+            .put(ENCODING, encoding) // data encoding
+            .build();
```

Comment:
Yeah, I guess the attributes like service name, env, language etc go here.

Reply:
There is a bunch of attributes that are added by default. Here we need to agree which attribute is used to tell that data contains allocation info vs cpu stack and how to specify whether the data is plain text or pprof.

Reply:
Afaiu, service name, env and language will be regular common otel resource attributes, so no need to add them into pprof.

Reply:
So it's `com.splunk.datatype` and `com.splunk.encoding`. Is it common in custom otlp data otlp to prefix the custom attributes by `com.splunk.`? As for the names, current ones are alright with me, perhaps `datatype -> profilerType` and `encoding -> format` would be a bit less ambiguous. Encoding makes me immediately think of UTF-8 :)

Reply:
Change to `com.splunk.profiling.data.type` and `com.splunk.profiling.data.format`. The `com.splunk.` prefix was already used for existing keys so just followed the same style. I guess it isn't really needed.

Reply:
Resolve